packages/cli/test/unit/commands/target.test.ts

```diff
@@ -0,0 +1,138 @@
+import ms from 'ms';
+import { describe, expect, it } from 'vitest';
+import { client } from '../../mocks/client';
+import { useUser } from '../../mocks/user';
+import target from '../../../src/commands/target';
+import { defaultProject, useProject } from '../../mocks/project';
+import { setupUnitFixture } from '../../helpers/setup-unit-fixture';
+import { useTeams } from '../../mocks/team';
+import createLineIterator from 'line-async-iterator';
+import { parseSpacedTableRow } from '../../helpers/parse-table';
+
+describe('target', () => {
+  it('should reject invalid arguments', async () => {
+    client.setArgv('--invalid');
+    await expect(target(client)).rejects.toThrow(
+      'unknown or unexpected option: --invalid'
+    );
+  });
+
+  describe('ls', () => {
+    it('should show custom environments with `vc target ls`', async () => {
+      useUser();
+      const teams = useTeams('team_dummy');
+      const team = Array.isArray(teams) ? teams[0] : teams.teams[0];
+      const { project } = useProject({
+        ...defaultProject,
+        name: 'static',
+        id: 'static',
+        customEnvironments: [
+          {
+            id: 'env_8DTiPYD33Rcvu2hQwYAdw0rwLquY',
+            slug: 'her',
+            createdAt: 1717176548879,
+            updatedAt: 1717176548879,
+            type: 'preview',
+            description: '',
+            name: 'her',
+            branchMatcher: {
+              type: 'endsWith',
+              pattern: 'her',
+            },
+          },
+          {
+            id: 'env_ph1tjPP20xp8VAuiFsYt4rhRYGys',
+            slug: 'ano',
+            createdAt: 1717176506341,
+            updatedAt: 1717176506341,
+            type: 'preview',
+            description: '',
+            name: 'ano',
+            branchMatcher: {
+              type: 'startsWith',
+              pattern: 'ano',
+            },
+          },
+        ],
+      });
+      client.cwd = setupUnitFixture('commands/deploy/static');
+      client.stderr.isTTY = false;
+      client.setArgv('target', 'ls');
+      const exitCode = await target(client);
+      expect(exitCode).toEqual(0);
+
+      const lines = createLineIterator(client.stderr);
+
+      let line = await lines.next();
+      expect(line.value).toEqual('Retrieving project…');
+
+      line = await lines.next();
+      expect(line.value).toEqual(
+        `Fetching custom environments for ${team.slug}/${project.name}`
+      );
+
+      line = await lines.next();
+      expect(line.value).contains(
+        `> 5 Custom Environments found under ${team.slug}/${project.name}`
```

Comment:
Blocking: Maybe it makes more sense to say "2 Custom Environments" or "5 Environments"?